randovania/games/prime_hunters/exporter/patch_data_factory.py

```diff
@@ -57,11 +61,34 @@ class HuntersPatchDataFactory(PatchDataFactory[HuntersConfiguration, HuntersCosm
     def game_enum(self) -> RandovaniaGame:
         return RandovaniaGame.METROID_PRIME_HUNTERS
 
-    def _calculate_starting_inventory(self, resources: ResourceCollection) -> dict[str, str]:
-        result = {
-            # TODO: Take starting weapons into account
-            "weapons_string": "00000101",
-        }
+    def _calculate_starting_inventory(self, resources: ResourceCollection) -> dict[str, str | int]:
+        result: dict[str, str | int] = {}
+
+        starting_items: defaultdict = defaultdict(int)
+        starting_items.update({resource.long_name: quantity for resource, quantity in resources.as_resource_gain()})
+
+        def starts_with_weapon(weapon_name: str) -> bool:
+            return starting_items[weapon_name] > 0
+
+        weapons = [
+            starts_with_weapon("Shock Coil"),
+            starts_with_weapon("Magmaul"),
+            starts_with_weapon("Judicator"),
+            starts_with_weapon("Imperialist"),
+            starts_with_weapon("Battlehammer"),
+            True,  # Missiles
+            starts_with_weapon("Volt Driver"),
+            True,  # Power Beam
+        ]
+        fmt = "{:d}" * 8  # 8-bit bitfield
+
+        result["weapons"] = fmt.format(*weapons)
+        result["missiles"] = starting_items["Missiles"]
+        result["ammo"] = 40
+
+        if starting_items["Energy Tank"] > 0:
```

Comment:
why the conditional?

Reply:
I figured if I didn't have to include it, it'd be fine but I'll revert it.